app/controllers/concerns/microsoft_concern.rb

```diff
@@ -5,8 +5,8 @@ def microsoft_client
     ::OAuth2::Client.new(ENV.fetch('AZURE_APP_ID', nil), ENV.fetch('AZURE_APP_SECRET', nil),
                          {
                            site: 'https://login.microsoftonline.com',
-                           authorize_url: 'https://login.microsoftonline.com/common/oauth2/v2.0/authorize',
-                           token_url: 'https://login.microsoftonline.com/common/oauth2/v2.0/token'
+                           authorize_url: "https://login.microsoftonline.com/#{azure_tenant_id}/oauth2/v2.0/authorize",
```

Comment:
Don't we need the above check

```
if ENV.fetch('AZURE_TENANT_ID', false)
```

To fix the authorization URL, will it be with AZURE_TENANT_ID or not? We should have the check.

Reply:
`azure_tenant_id` returns the tenant ID if it is set, otherwise it defaults to `common`, which is what is used for multi tenant. No check required.